### PR TITLE
Refactor/negotiation trajectory

### PR DESCRIPTION
**Pull request by ChawinTan**
## What's new

#206 - following the discussion on removing negotiation trajectories in the leaflet overlay checkbox

https://user-images.githubusercontent.com/29136440/106719686-91cb7380-663d-11eb-8367-4687f8b629cd.mp4

- removed the negotiations trajectory option in the leaflet overlay checkbox
- only show negotiation trajectory when user is in the negotiations panel and hide the normal trajectories
- when user navigates out of the negotiations panel, go back to showing normal trajectories

Showing the different trajectories is done by passing the current index of the omni panel to visual schedulizer. If the index matches the index of negotiations panel, the existing trajectories will be cleared.

Two other minor issues:
- home button currently requires 2 clicks to go back to current main menu. 
- the close button currently still opens to existing view despite calling `resetView`.

Solved that by creating a new object in `dashboard reducer` with the current view set to the result of `stack.stackNavigator.reset()`

Right now, users still have to click on the negotiation to show the negotiations trajectory. Is there any intention to automatically select the latest accepted negotiations that removes the need for clicking?

## Self-checks

- [x] I'm familiar with and follow this [Typescript guideline](https://basarat.gitbook.io/typescript/styleguide)
- [ ] I added unit-tests for new components
- [x] I tried testing edge cases
- [x] I tested the behavior of the components that interact with the backend, with an e2e test

### CODE REVIEW

**Comment by koonpeng** [comment]
Thanks, I like how it looks, does this PR also fix the new loop and delivery request changes?

> Two other minor issues:
>
> home button currently requires 2 clicks to go back to current main menu.
> the close button currently still opens to existing view despite calling resetView.

These are non related issues right? I guess they are introduced when I did the schedule visualizer/dashboard refactor.

**Comment by ChawinTan** [comment]
> Thanks, I like how it looks, does this PR also fix the new loop and delivery request changes?
> 
> >

Nope, I am on an old version of RMF demos



> These are non related issues right? I guess they are introduced when I did the schedule visualizer/dashboard refactor.

Yup, they are not related. But the UI does not look in sync so I guess they are indirectly related

File: packages/dashboard/src/components/schedule-visualizer/index.tsx [hunk 1]
```diff
@@ -195,17 +196,22 @@ export default function ScheduleVisualizer(props: ScheduleVisualizerProps): Reac
           },
         });
         debug('set trajectories');
-        setTrajectories((prev) => ({
-          ...prev,
-          [curMapFloorLayer.level.name]: resp,
-        }));
+        // negotiations panel index is 6 on the omni panel
+        if (omniPanelView === 6) {
```
Comment (matiasbavera) [inline packages/dashboard/src/components/schedule-visualizer/index.tsx:200]: Can we use some sort of enum here to avoid hardcoding the index position?

**Review by matiasbavera — COMMENTED** [review]
Thanks for these changes, looks good. Left a comment.

File: packages/dashboard/src/components/schedule-visualizer/index.tsx (hunk 1 above)
Comment (ChawinTan) [reply packages/dashboard/src/components/schedule-visualizer/index.tsx]: Re-factored out the enum

File: packages/dashboard/src/components/schedule-visualizer/index.tsx [hunk 2]
```diff
@@ -43,6 +44,7 @@ export interface MapFloorLayer {
 export interface ScheduleVisualizerProps extends React.PropsWithChildren<{}> {
   buildingMap: RomiCore.BuildingMap;
   negotiationTrajStore: Readonly<Record<string, NegotiationTrajectoryResponse>>;
+  omniPanelView: number;
```
Comment (koonpeng) [inline packages/dashboard/src/components/schedule-visualizer/index.tsx:47]: This coupled it with the omnipanel, instead, leave the logic whether to show/hide trajectories to `Dashboard`, which has knowledge of both `OmniPanel` and `ScheduleVisualizer` to make that decision.

Comment (ChawinTan) [reply packages/dashboard/src/components/schedule-visualizer/index.tsx]: Used a boolean check at the dashboard instead

File: packages/dashboard/src/components/schedule-visualizer/index.tsx [hunk 3]
```diff
@@ -44,7 +43,7 @@ export interface MapFloorLayer {
 export interface ScheduleVisualizerProps extends React.PropsWithChildren<{}> {
   buildingMap: RomiCore.BuildingMap;
   negotiationTrajStore: Readonly<Record<string, NegotiationTrajectoryResponse>>;
-  omniPanelView: number;
+  showTrajectories: boolean;
```
Comment (koonpeng) [inline packages/dashboard/src/components/schedule-visualizer/index.tsx:46]: Suggest to make this optional and default to `true`.

**Review by koonpeng — CHANGES_REQUESTED** [review]

File: packages/dashboard/src/components/schedule-visualizer/index.tsx (hunk 3 above)
Comment (ChawinTan) [reply packages/dashboard/src/components/schedule-visualizer/index.tsx]: set it to optional

Added a check for showTrajectories === undefined which will result in the default behaviour when showTrajectory is true

**Review by koonpeng — APPROVED** [review]
